docs/operator-manual/argocd-cm.yaml

```
@@ -429,4 +429,8 @@ data:
   webhook.maxPayloadSizeMB: "50"
 
   # application.sync.impersonation.enabled enables application sync to use a custom service account, via impersonation. This allows decoupling sync from control-plane service account.
-  application.sync.impersonation.enabled: "false"
\ No newline at end of file
+  application.sync.impersonation.enabled: "false"
+
+  # Set this flag to false to revert to old behavior of allowing to pass different revision from the one given in the application when syncing.
+  application.sync.externalRevisionConsideredOverride: "true"
```

I would reword this in a way to say that `false` is the current default, but it's highly recommended to set this to `true` and that the next major release will flip this flag. Something like 

```suggestion
  # The current behavior allows passing a different revision from the one given in the application when syncing. We highly recommend that this be set to `true`. The next major release will set the default to be `false`.
  application.sync.externalRevisionConsideredOverride: "false"
```